### ISSUE TITLE
Convert UUID to String

### ISSUE THREAD

**Issue by andatt** [yandex/ClickHouse#4362]
I am using the function:

generateUUIDv4()

How can I convert the result of this to string?

Have tried UUIDStringToNum, toString, toFixedString. None of these work.

Thanks

**Comment by hatarist** [comment]
Weird. `toString` does work for me:

```
 :) CREATE TABLE issue4362 (uuid UUID, string String) Engine = Memory();

 :) INSERT INTO issue4362 SELECT generateUUIDv4() uuid, toString(uuid);

 :) INSERT INTO issue4362 SELECT generateUUIDv4() uuid, toString(uuid);

 :) INSERT INTO issue4362 SELECT generateUUIDv4() uuid, toString(uuid);

 :) SELECT * FROM issue4362;

┌─────────────────────────────────uuid─┬─string───────────────────────────────┐
│ a01eb025-2a25-4ddf-aebe-83ed7e171a7a │ a01eb025-2a25-4ddf-aebe-83ed7e171a7a │
│ 87c94153-099d-49c1-920d-8c6f072c08c6 │ 87c94153-099d-49c1-920d-8c6f072c08c6 │
│ b914e23d-1838-4015-847b-49539316db98 │ b914e23d-1838-4015-847b-49539316db98 │
└──────────────────────────────────────┴──────────────────────────────────────┘

```

How do you try and what does Clickhouse tell you?

**Comment by andatt** [comment]
Hi thanks!

I tried posting the queries above and I got this:

```
INSERT INTO issue4362 SELECT 
    generateUUIDv4() AS uuid, 
    toString(uuid)

Received exception from server (version 18.10.3):
Code: 43. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: UInt128 cannot be write as a text. 

0 rows in set. Elapsed: 0.133 sec. 

d2d1f6d0b1cd :) 
```
Looks like my version is older, is that the cause of the issue?

**Comment by etah000** [comment]
also works on 18.14.13.

**Comment by andatt** [comment]
Using the 18.10.3 docker container from https://hub.docker.com/r/yandex/clickhouse-server/tags

this is what I get:

```
(srf_framework_venv) andrew@andrew-Inspiron-7560:~/python/srframework/senseon-reasoning-framework$ docker exec -it -u root some-clickhouse-server_18103 bash
root@569ff78b093b:/# clickhouse --client
ClickHouse client version 18.10.3.
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 18.10.3 revision 54405.

569ff78b093b :) CREATE TABLE issue4362 (uuid UUID, string String) Engine = Memory();

CREATE TABLE issue4362
(
    uuid UUID, 
    string String
)
ENGINE = Memory()

Ok.

0 rows in set. Elapsed: 0.009 sec. 

569ff78b093b :) INSERT INTO issue4362 SELECT generateUUIDv4() uuid, toString(uuid);

INSERT INTO issue4362 SELECT 
    generateUUIDv4() AS uuid, 
    toString(uuid)

Received exception from server (version 18.10.3):
Code: 43. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: UInt128 cannot be write as a text. 

0 rows in set. Elapsed: 0.104 sec. 

569ff78b093b :) 
```

**Comment by nezed** [comment]
Was fixed somewhere between `18.12.13` and `18.14.13`

So the only solution is to upgrade

**Comment by andatt** [comment]
Thanks. That explains it!
